components/Connect.tsx

```diff
@@ -70,6 +70,46 @@ export default function BluetoothConnect({ bikeCredentials, setBikeInstance, bac
                     Back to login
                 </Button>
             </P>
+
+            <P vertical={10}>
+                <Button
+                    onClick={() => {
+                        const blob = new Blob([JSON.stringify(bikeCredentials)], { type: 'application/json' })
+                        const url = URL.createObjectURL(blob)
+
+                        // If the user is on iOS and has the Bluefy browser, we can't save the file directly
+                        // so we'll just show the user the JSON file and they can save it manually
+                        if (navigator.userAgent.includes('iPhone') && navigator.userAgent.includes('Bluefy')) {
```

Comment:
This feels a bit hacky.
How about always showing the credentials on screen when clicking this button and also downloading them.
That way the user will see a file is downloaded thus not needing to copy the contents on screen and for browsers that for some wired reason can't download files they can copy the contents on screen.

Reply:
It is pretty big especially for mobile devices hence I preferred to only showing it when necessarily but will do till we come up with a better idea maybe a like box that can be minimised and with just a copy button

Reply:
You can make the font size smaller and place it within a box with scrollbars maybe?
And add a button to copy the credentials to the clipboard maybe?

Reply:
There is a copy button already?